addons/web_editor/static/src/js/snippets.options.js

```diff
@@ -580,7 +580,7 @@ odoo.define('web_editor.snippets.options', function (require) {
                         .on("background-color-event.background-option", (function (e, type) {
                             e.stopPropagation();
                             if (e.currentTarget !== e.target) return;
-                            this.$el.find("li:first > a").trigger(type);
+                            this.$el.find("li[data-background=''] > a").trigger(type);
```

Comment:
'Set Custom image' is swaped with 'None' in snippets background option; now 'None' is on second position in background option not at the first; while background color is set  we have to set none as a background image. so now it will identify from the attribute  data-background=' ', not as a li:first